packages/calcite-components/src/components/dropdown/dropdown.e2e.ts

```diff
@@ -847,6 +847,36 @@ describe("calcite-dropdown", () => {
         expect(calciteDropdownOpen).toHaveReceivedEventTimes(2);
         expect(calciteDropdownClose).toHaveReceivedEventTimes(1);
       });
+
+      it("toggles when Enter keydown is dispatched", async () => {
+        const page = await newE2EPage();
+        await page.setContent(html`
+          <calcite-dropdown>
+            <calcite-button slot="trigger">Open dropdown</calcite-button>
+            <calcite-dropdown-group selection-mode="single">
+              <calcite-dropdown-item id="item-1"> Dropdown Item Content </calcite-dropdown-item>
+              <calcite-dropdown-item id="item-2" selected> Dropdown Item Content </calcite-dropdown-item>
+            </calcite-dropdown-group>
+          </calcite-dropdown>
+        `);
+        const element = await page.find("calcite-dropdown");
+        const dropdownWrapper = await page.find(`calcite-dropdown >>> .calcite-dropdown-wrapper`);
+        const calciteDropdownOpen = await element.spyOnEvent("calciteDropdownOpen");
+        const calciteDropdownClose = await element.spyOnEvent("calciteDropdownClose");
+        const waitForCalciteDropdownOpen = page.waitForEvent("calciteDropdownOpen");
+
+        expect(await dropdownWrapper.isVisible()).toBe(false);
+
+        await page.$eval("calcite-button[slot='trigger']", (triggerEl: HTMLCalciteButtonElement) => {
+          triggerEl.dispatchEvent(new KeyboardEvent("keydown", { key: "Enter", bubbles: true }));
```

Comment:
Could you add a comment here about only emitting `keydown` to avoid emitting `click` on the button?

Reply:
If you wanted to stick to puppeteer APIs, I think you could also do the following here:

```tsx
const trigger = await page.find("calcite-button[slot='trigger']");
await trigger.callMethod("setFocus");
await page.keyboard.down("Enter"); // intentionally not pressing to avoid emitting `click`
```

Totally up to you.

Reply:
I think I like the dispatch event better since it targets the issue more accurately and doesn't require any focusing.